### PR TITLE
Add teal flavor

### PR DESCRIPTION
**Pull request by mudler**
Add teal flavor to our pipelines.

Part of https://github.com/rancher-sandbox/cOS-toolkit/issues/1238

Signed-off-by: Ettore Di Giacinto <edigiacinto@suse.com>

### CODE REVIEW

**Comment by mudler** [comment]
I'd go ahead with this and check out manually the artifacts to see if there are any issues and will double check with the ones we currently have for green, I bet there will be some small adjustments to do

File: packages/kernel/build.yaml [hunk 1]
```diff
@@ -3,6 +3,16 @@ requires:
   category: "distro"
   version: ">=0"
 
+excludes:
+- ^/boot/grub2
```
Comment (mudler) [inline packages/kernel/build.yaml:7]: I think those two previously just slipped in, no real need to add them as part of the kernel package

File: packages/utils/gomplate/build.yaml [hunk 2]
```diff
@@ -7,16 +7,9 @@ package_dir: /{{.Values.name}}
 
 env:
 - PATH=$PATH:/usr/local/go/bin
+- CGO_ENABLED=0
```
Comment (mudler) [inline packages/utils/gomplate/build.yaml:10]: somehow those slipped in :man_shrugging:

File: packages/kernel/build.yaml [hunk 3]
```diff
@@ -3,6 +3,16 @@ requires:
...
+- ^/boot/initrd
+{{ if .Values.codename }}
+{{if eq .Values.codename "teal" }}
+unpack: true
+includes:
+- ^/boot|^/boot/.*
```
Comment (mudler) [inline packages/kernel/build.yaml:13]: The real difference in teal is that we extract all the kernels from the image, so no zypper in or build happening

File: packages/templates/distro.yaml [hunk 4]
```diff
@@ -1,4 +1,6 @@
 {{ define "distro_install" }}
+## All packages for teal are in the teal image already.
+{{ if ne .Values.codename "teal" }}
```
Comment (mudler) [inline packages/templates/distro.yaml:3]: This is gating accidental calls to `zypper in..` if we are in `teal`. supposedly, we don't add any repo to it

**Review by mjura — APPROVED** [review]
LGTM

**Pull request merged**